src/rust/node-identifier/src/lib.rs

```diff
@@ -61,23 +44,29 @@ use sqs_executor::{
     s3_event_retriever::S3PayloadRetriever,
     time_based_key_fn,
 };
+use tap::tap::TapOptional;
 
-macro_rules! wait_on {
-    ($x:expr) => {{
-        $x.await
-    }};
-}
+use crate::error::NodeIdentifierError;
 
 pub mod dynamic_sessiondb;
-
+mod error;
 pub mod sessiondb;
 pub mod sessions;
 
+/**
+    The `NodeIdentifier` takes in graphs of previously unidentified nodes and performs identification
+    based on the configured strategies for that node type.
+
+    The strategies come in two variants:
+
+    * [Session](`grapl_graph_descriptions::graph_description::Session`) - strategy used for nodes with lifetimes (e.g. process, network connection)
+    * [Static](`grapl_graph_descriptions::graph_description::Static`) - strategy used for nodes with canonical and unique identifiers (e.g. aws events)
+*/
 #[derive(Clone)]
 pub struct NodeIdentifier<D, CacheT>
 where
-    D: DynamoDb + Clone + Send + Sync + 'static,
-    CacheT: Cache + Clone + Send + Sync + 'static,
+    D: DynamoDb + Clone + Send + Sync,
+    CacheT: Cache + Clone + Send + Sync,
```

Comment:
A) I believe there's some way to deduplicate all these constraints throughout this file
B) If so - should we?

Reply:
Good call. That's probably a good idea; i forgot I could use `type <some_name> = <trait bounds>` to simply this.

Reply:
err, not type aliases. I think I can do this with a new Trait that requires the appropriate trait bounds. Hmm Let me experiment a bit more.

Reply:
```
pub trait EventHandlerDynamoDb: DynamoDb + Clone + Send + Sync {}
pub trait EventHandlerCache: Cache + Clone + Send + Sync {}

impl<D> EventHandlerDynamoDb for D where D: DynamoDb + Clone + Send + Sync {}
impl<C> EventHandlerCache for C where C: Cache + Clone + Send + Sync {}
```

Seems to work. I'll add this down into the sqs-executor so we get this everywhere.